src/ContaoCommunityAlliance/DcGeneral/Contao/Dca/Builder/Legacy/LegacyDcaDataDefinitionBuilder.php

```diff
@@ -1280,7 +1282,14 @@ protected function parseProperties(ContainerInterface $container)
             if ($definition->hasProperty($propName)) {
                 $property = $definition->getProperty($propName);
             } else {
-                $property = new DefaultProperty($propName);
+                if ((true === array_key_exists('collection', $propInfo['eval']))
+                    ||(true === array_key_exists('columnFields', $propInfo['eval']))
+                    ||(true === array_key_exists('columnsCallback', $propInfo['eval']))
+                ) {
+                    $property = new DefaultCompoundProperty($propName);
```

Comment:
For `collection`, this should add the sub properties here - for the other cases, leave it to MCW.

Reply:
change